freqtrade/freqtradebot.py

```diff
@@ -169,6 +169,7 @@ def process(self) -> None:
         with self._sell_lock:
             # Check and handle any timed out open orders
             self.check_handle_timedout()
+            self.check_handle_custom_entryprice_outdated()
```

Comment:
Please treat this as a seperate feature (which means a seperate PR).
Adding timeouts based on price should not be part of this in my opinion (as i mentioned to you on discord already some time ago).

i'm not totally against this feature per se - but i'll want to have this as a seperate, isolated PR - which makes it clear which parts are changed because of the custom entryprice - and which because of the "trailing price".

The 2 are not connected, really - and we can have either of the two without the other.
we can also timeout orders based on "pre-calculated" pricing (not custom price) without ever returning a custom price - but based on a % offset that price moved in the wrong direction.

As such - please revert the changes related to this handling.

Reply:
I see, but how we treat unfulfilled custom entry orders? Just with the timeout for the moment?

Reply:
I don't see timeout related to this PR.

Currently - we use `get_rate()` to define a price (maybe 4rd spot below current price in the orderbook - you can configure that).
with the outcome of this PR - the result of `get_rate()` is passed to the strategy - which has the ability to modify / completely change this result.
If it does - we'll simply have a new price - maybe between 3rd and 4th spot in the orderbook - maybe 0.5% below current price - maybe 200 (just for the fun of it).

This does not impact timout handling at all - and regular timeout rules will apply just as usual.

---

What you're trying to add here is trailing entry (where you can potentially follow the price if it moves away from your order).
This can be useful with custom pricing - but can at the same time be used with the regular `get_rate()` method (without ever defining a custom rate). 
As such - it's completely unrelated to specifying your own price. 
Having the 2 seperate will clearly show which parts of the code are changed because of which.

Before investigating / investing too much time into this - please also check / evaluate if this can't be handled by `check_buy_timeout` and `check_sell_timeout` respectively.
If you look at [the samples](https://www.freqtrade.io/en/latest/strategy-advanced/#trade-entry-buy-order-confirmation) - it's basically doing exactly this - cancelling the order if price moved away from the current order.

That will give the strategy full control over it - while providing you basically the same functionality (you can use `trade.open_rate_requested` to get the original price of entry).